src/core/agent/services/ipexCommunicationService.test.ts

```diff
@@ -983,6 +983,153 @@ describe("Ipex communication service of agent", () => {
     });
   });
 
+  test("Cannot get linkedGroupRequest from ipex/apply if the notification is missing in the DB", async () => {
+    const id = "uuid";
+    const date = new Date().toISOString();
+    const notification = {
+      id,
+      createdAt: date,
+      a: {
+        d: "d",
+      },
+      connectionId: "EGR7Jm38EcsXRIidKDZBYDm_xox6eapfU1tqxdAUzkFd",
+      read: true,
+    };
+
+    notificationStorage.findById.mockResolvedValueOnce(null);
+
+    await expect(
+      ipexCommunicationService.getLinkedGroupFromIpexApply(notification.id)
+    ).rejects.toThrowError(
+      `${IpexCommunicationService.NOTIFICATION_NOT_FOUND} ${id}`
+    );
+  });
+
+  test("Should return accepted and membersJoined with each credential when linkedGroupRequests from ipex/apply contain valid data", async () => {
+    const id = "uuid";
+    const date = new Date().toISOString();
+    const notification = {
+      id,
+      createdAt: date,
+      a: {
+        d: "d",
+      },
+      connectionId: "EGR7Jm38EcsXRIidKDZBYDm_xox6eapfU1tqxdAUzkFd",
+      read: true,
+    };
+
+    const grantNoteRecord = {
+      linkedGroupRequests: {
+        credentialSaid1: {
+          accepted: true,
+          saids: {
+            ipexOfferSaid1: [
+              ["memberA", "multisigExn1A"],
+              ["memberB", "multisigExn1B"],
+            ],
+            ipexOfferSaid2: [["memberA", "multisigExn2A"]],
+          },
+        },
+        credentialSaid2: {
+          accepted: true,
+          saids: {
+            ipexOfferSaid1: [
+              ["memberC", "multisigExn1C"],
+              ["memberD", "multisigExn1D"],
```

Comment:
Can we remove this line?

That way we can be sure it's iterating over `ipexOfferSaid2` as well in the test. And we'll have covered all scenarios

Reply:
Do you mean to remove `ipexOfferSaid1` and keep `ipexOfferSaid2` in credentialSaid2?

Reply:
No, just remove 1038. So only `memberC` is in said1 and only `memberD` is in said2. Of this credential, the other credential can stay the same.

The reason is because if we had a bug where it only looked at `ipexOfferSaid1` and never the other SAIDs, this test wouldn't catch the bug because everyone in said2 is already in said1.